From c7e8b0dfe7469d2a59cb9d211d43236b18c64437 Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Mon, 31 Oct 2011 17:58:44 +0800
Subject: [PATCH 051/641] UniCore64: Add functions show_stack and trap_init

Fixed error:
elfcore.c:(.text+0x35e0): undefined reference to `show_stack'
calibrate.c:(.init.text+0xaf8): undefined reference to `trap_init'

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/Makefile |    2 +-
 arch/unicore64/kernel/traps.c  |   22 ++++++++++++++++++++++
 2 files changed, 23 insertions(+), 1 deletion(-)
 create mode 100644 arch/unicore64/kernel/traps.c

diff --git a/arch/unicore64/kernel/Makefile b/arch/unicore64/kernel/Makefile
index e8a961f..ea2c5c0 100644
--- a/arch/unicore64/kernel/Makefile
+++ b/arch/unicore64/kernel/Makefile
@@ -1,4 +1,4 @@
 obj-y			:= ptrace.o setup.o init_task.o process.o
-obj-y			+= time.o
+obj-y			+= time.o traps.o
 head-y			:= head.o
 extra-y			:= $(head-y) vmlinux.lds
diff --git a/arch/unicore64/kernel/traps.c b/arch/unicore64/kernel/traps.c
new file mode 100644
index 0000000..5bdc260
--- /dev/null
+++ b/arch/unicore64/kernel/traps.c
@@ -0,0 +1,22 @@
+#include <linux/kernel.h>
+#include <linux/init.h>
+
+/**
+ * trap_init() -
+ */
+void __init trap_init(void)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * show_stack() -
+ * @tsk:
+ * @sp:
+ */
+void show_stack(struct task_struct *tsk, unsigned long *sp)
+{
+	/* FIXME */
+	BUG();
+}
-- 
1.7.9.5

